core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -119,13 +123,71 @@
     private String serviceLoggerPrefix = DEFAULT_SERVICE_LOGGER_PREFIX;
     private Consumer<RequestLog> accessLogWriter = AccessLogWriters.disabled();
 
+    @Nullable
     private Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>> decorator;
 
+    /**
+     * Adds an HTTP port that listens on all available network interfaces. If no HTTP(S) port is added
+     * (i.e. neither {@code http()} nor {@code https()} method is called), a default of {@code 0}
+     * (randomly-assigned port) will be used.
+     *
+     * @param port the HTTP port number.
+     *
+     * @see #http(InetSocketAddress)
+     */
+    public ServerBuilder http(int port) {
+        ports.add(new ServerPort(port, HTTP));
+        return this;
+    }
+
+    /**
+     * Adds an HTTP port that listens to the specified {@code localAddress}. If no HTTP(S) port is added
+     * (i.e. neither {@code http()} nor {@code https()} method is called), a default of {@code 0}
+     * (randomly-assigned port) will be used.
+     *
+     * @param localAddress the local address to bind
+     *
+     * @see #http(int)
+     */
+    public ServerBuilder http(InetSocketAddress localAddress) {
+        ports.add(new ServerPort(requireNonNull(localAddress, "localAddress"), HTTP));
+        return this;
+    }
+
+    /**
+     * Adds an HTTPS port that listens on all available network interfaces. If no HTTPS port is added,
+     * (i.e. no {@code https()} method is called), HTTPS support is disabled.
```

Comment:
Just curious. What if a user called `tls()` method? I'm not sure but what's better behavior? Listening on randomly-assigned HTTPS port or disabling HTTPS or raising an exception? (Actually the current behavior is also OK to me)

Reply:
That's a good question. Refactored a little bit to default to HTTPS when tls() was called and added documentation about the behavior.